wandb/sdk/launch/agent/agent.py

```diff
@@ -163,23 +164,33 @@ def loop(self) -> None:
             while True:
                 self._ticks += 1
                 job = None
-                for queue in self._queues:
-                    job = self.pop_from_queue(queue)
-                    if job:
-                        break
+                if self._running < self._max_jobs:
+                    for queue in self._queues:
+                        job = self.pop_from_queue(queue)
+                        if job:
+                            break
                 if not job:
                     time.sleep(AGENT_POLLING_INTERVAL)
+                    agent_response = self._api.get_launch_agent(
+                        self._id, self.gorilla_supports_agents
+                    )
+                    self._name = agent_response[
+                        "name"
+                    ]  # hacky, but we don't return the name on create so this is first time
+                    if agent_response["stopPolling"]:
```

Comment:
this should only happen when there are no jobs running. We don't want to interrupt jobs by hitting stop.